include/faabric/scheduler/Scheduler.h

```diff
@@ -220,6 +235,27 @@ class Scheduler
 
     ExecGraph getFunctionExecGraph(unsigned int msgId);
 
+    // ----------------------------------
+    // Function Migration
+    // ----------------------------------
+    void checkForMigrationOpportunities();
+
+    std::shared_ptr<faabric::PendingMigrations> getPendingAppMigrations(
+      uint32_t appId);
+
+    void addPendingMigration(std::shared_ptr<faabric::PendingMigrations> msg);
+
+    void removePendingMigration(uint32_t appId);
+
+    // ----------------------------------
+    // Clients
+    // ----------------------------------
+    faabric::scheduler::FunctionCallClient& getFunctionCallClient(
+      const std::string& otherHost);
+
+    faabric::snapshot::SnapshotClient& getSnapshotClient(
```

Comment:
We need the snapshot client to be a public member in faasm/faasm#565 before migrating a function. In that case we are doing a function chain between a function in a non-master host, and a function in the master host. Faabric won't register,take, and push snapshots in this case; we have to do it ourselves manually. Thus, need access to the snapshot client.

We don't need the function call client to be a public member, but I was reluctant to split the declaration of the clients (happy to revert).